test/parallel/test-http-hostname-typechecking.js

```diff
@@ -21,7 +21,7 @@ vals.forEach((v) => {
 // These values are OK and should not throw synchronously
 ['', undefined, null].forEach((v) => {
   assert.doesNotThrow(() => {
-    http.request({hostname: v}).on('error', common.noop).end();
-    http.request({host: v}).on('error', common.noop).end();
+    http.request({hostname: v}).on('error', common.mustCall()).end();
+    http.request({host: v}).on('error', common.mustCall()).end();
```

Comment:
I think I ran into an issue like this before; the thing is, this is not guaranteed to fail, since the machine running the tests may have an HTTP server running on port 80.

@Trott Any suggestions? Can/should we revert this? The next-best alternative would be trying to allocate a port by creating a server, then closing it and trying to use that as the por value (or actually run an HTTP server that serves proper responses?).